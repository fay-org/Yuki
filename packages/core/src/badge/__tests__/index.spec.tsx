import { mount } from '@vue/test-utils'
import Badge from '..'

describe('Badge', () => {
  it('render normal', () => {
    const wrapper = mount(Badge)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('type', () => {
    const wrapper = mount({
      render() {
        return (
          <div>
            <Badge scale={2 / 3} type="default" />
            <Badge type="secondary" />
            <Badge type="success" />
            <Badge type="warning" />
            <Badge type="error" />
          </div>
        )
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('dot', () => {
    const wrapper = mount(Badge, {
      props: {
        dot: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
