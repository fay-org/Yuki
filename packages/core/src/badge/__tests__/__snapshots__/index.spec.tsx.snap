// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge dot 1`] = `"<span class="fect-badge fect-badge--default dot" style="--badge-color: #fff; --badge-bg-color: #000; --badge-font-size: calc(0.875 * 16px); --badge-width: auto; --badge-height: auto; --badge-pt: calc(0.25 * 16px); --badge-pr: calc(0.4375 * 16px); --badge-pb: calc(0.25 * 16px); --badge-pl: calc(0.4375 * 16px); --badge-mt: 0; --badge-mr: 0; --badge-mb: 0; --badge-ml: 0; --badge-dot-px: calc(0.25 * 16px); --badge-dot-py: calc(0.25 * 16px);"></span>"`;

exports[`Badge type 1`] = `"<div><span class="fect-badge fect-badge--default" style="--badge-color: #fff; --badge-bg-color: #000; --badge-font-size: calc(0.7291666666666666 * 16px); --badge-width: auto; --badge-height: auto; --badge-pt: calc(0.20833333333333331 * 16px); --badge-pr: calc(0.3645833333333333 * 16px); --badge-pb: calc(0.20833333333333331 * 16px); --badge-pl: calc(0.3645833333333333 * 16px); --badge-mt: 0; --badge-mr: 0; --badge-mb: 0; --badge-ml: 0; --badge-dot-px: calc(0.20833333333333331 * 16px); --badge-dot-py: calc(0.20833333333333331 * 16px);"><!----></span><span class="fect-badge fect-badge--secondary" style="--badge-color: white; --badge-bg-color: #666; --badge-font-size: calc(0.875 * 16px); --badge-width: auto; --badge-height: auto; --badge-pt: calc(0.25 * 16px); --badge-pr: calc(0.4375 * 16px); --badge-pb: calc(0.25 * 16px); --badge-pl: calc(0.4375 * 16px); --badge-mt: 0; --badge-mr: 0; --badge-mb: 0; --badge-ml: 0; --badge-dot-px: calc(0.25 * 16px); --badge-dot-py: calc(0.25 * 16px);"><!----></span><span class="fect-badge fect-badge--success" style="--badge-color: white; --badge-bg-color: #0070f3; --badge-font-size: calc(0.875 * 16px); --badge-width: auto; --badge-height: auto; --badge-pt: calc(0.25 * 16px); --badge-pr: calc(0.4375 * 16px); --badge-pb: calc(0.25 * 16px); --badge-pl: calc(0.4375 * 16px); --badge-mt: 0; --badge-mr: 0; --badge-mb: 0; --badge-ml: 0; --badge-dot-px: calc(0.25 * 16px); --badge-dot-py: calc(0.25 * 16px);"><!----></span><span class="fect-badge fect-badge--warning" style="--badge-color: white; --badge-bg-color: #f5a623; --badge-font-size: calc(0.875 * 16px); --badge-width: auto; --badge-height: auto; --badge-pt: calc(0.25 * 16px); --badge-pr: calc(0.4375 * 16px); --badge-pb: calc(0.25 * 16px); --badge-pl: calc(0.4375 * 16px); --badge-mt: 0; --badge-mr: 0; --badge-mb: 0; --badge-ml: 0; --badge-dot-px: calc(0.25 * 16px); --badge-dot-py: calc(0.25 * 16px);"><!----></span><span class="fect-badge fect-badge--error" style="--badge-color: white; --badge-bg-color: #e00; --badge-font-size: calc(0.875 * 16px); --badge-width: auto; --badge-height: auto; --badge-pt: calc(0.25 * 16px); --badge-pr: calc(0.4375 * 16px); --badge-pb: calc(0.25 * 16px); --badge-pl: calc(0.4375 * 16px); --badge-mt: 0; --badge-mr: 0; --badge-mb: 0; --badge-ml: 0; --badge-dot-px: calc(0.25 * 16px); --badge-dot-py: calc(0.25 * 16px);"><!----></span></div>"`;
